command/server.go

```diff
@@ -1223,6 +1275,57 @@ func (c *ServerCommand) enableThreeNodeDevCluster(base *vault.CoreConfig, info m
 	return 0
 }
 
+// addPlugin adds any plugins to the catalog
+func (c *ServerCommand) addPlugin(token string, core *vault.Core) error {
+	// Get the sha256 of the file at the given path.
+	pluginSum := func(p string) (string, error) {
+		hasher := sha256.New()
+		f, err := os.Open(p)
+		if err != nil {
+			return "", err
+		}
+		defer f.Close()
+		if _, err := io.Copy(hasher, f); err != nil {
+			return "", err
+		}
+		return hex.EncodeToString(hasher.Sum(nil)), nil
+	}
+
+	// Mount any test plugins. We do this explicitly before we inform tests of
+	// a completely booted server intentionally.
+	sha256sum, err := pluginSum(c.flagDevPluginPath)
+	if err != nil {
+		return err
+	}
+
+	// Default the name to the basename of the binary
```

Comment:
It seems a bit unnecessary for testing to need to specify a directory, a path, a name, and a command.

Given that the plugin dir is already a flag, if we standardize on the name being the name of the binary and the command being the name of the binary, then it's trivial to use kv-builder to be able to set args per-plugin. This would allow multiple plugins to be auto loaded in a predictable way.

Reply:
Hmm okay - do you want me to update the `-dev-plugin-dir` flag to add all the plugins in the given directory? 

I'm not sure what you mean by 

> then it's trivial to use kv-builder to be able to set args per-plugin

though. Can you show an example of what you're thinking that CLI would be?

Reply:
Ah I see. Hmm what are you thoughts on:

1. "Upgrading" `-dev-plugin-dir` to do this automatically (instead of adding a -path flag) and then adding a `-dev-plugin-enable` flag that defaults to true, but can be turned off when not needed

1. Ignoring args and other commands in this PR, and fixing it when we need them? (So just default to basename as the command with no args)

Reply:
I think that's fine. Also many (most?) plugins can/should be able to use env vars in addition to args, so that seems like a relatively easy way to test without having to add more flags, if args are even needed.

Reply:
Okay @jefferai updated